src/components/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,86 @@
+import type React from 'react'
+
+import PropTypes from 'prop-types'
+import { Button } from 'components/ui/button'
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  DropdownMenuSeparator
+} from 'components/ui/dropdown-menu'
+import { ChevronDown } from 'lucide-react'
+
+interface MultiSelectDropdownProps {
+  selectedItems: string[];
+  items: string[];
+  onItemToggle: (item: string) => void;
+  placeholder?: string;
+  allItemsLabel: string;
+}
+
+const MultiSelectDropdown: React.FC<MultiSelectDropdownProps> = ({
+                                                                   selectedItems,
+                                                                   items,
+                                                                   onItemToggle,
+                                                                   placeholder,
+                                                                   allItemsLabel
+                                                                 }) => {
+  const allSelected = selectedItems.includes(allItemsLabel)
+
+  let displayValue: string
+  if (allSelected) {
+    displayValue = allItemsLabel
+  } else if (selectedItems.length === 1) {
+    displayValue = selectedItems[0]
+  } else if (selectedItems.length > 1) {
+    displayValue = `${selectedItems.length} selected`
+  } else {
+    displayValue = placeholder ?? 'Select items'
+  }
+
+ const handleItemClick = (item: string, event: Event) => {
+    event.preventDefault();
+    onItemToggle(item);
+  };
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button variant="outline" className="w-full justify-between"
+                aria-label={`Select ${allItemsLabel.toLowerCase()}`}>
+          <span className="text-dropdown-text font-normal"> {displayValue} </span>
+          <ChevronDown className="h-4 w-4" />
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="w-full">
+        <DropdownMenuItem
+          onSelect={e => handleItemClick(allItemsLabel, e)}
+          className={allSelected ? 'bg-accent' : ''}
+        >
+          {allItemsLabel}
+        </DropdownMenuItem>
+        <DropdownMenuSeparator />
+        {items.filter(item => item !== allItemsLabel).map(item => (
+          <DropdownMenuItem
+            key={item}
+            onSelect={e => handleItemClick(item, e)}
+            className={selectedItems.includes(item) ? 'bg-accent' : ''}
+          >
+            {item}
+          </DropdownMenuItem>
+        ))}
+      </DropdownMenuContent>
+    </DropdownMenu>
+  )
+}
+
+MultiSelectDropdown.propTypes = {
+  selectedItems: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+  items: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+  onItemToggle: PropTypes.func.isRequired,
+  placeholder: PropTypes.string,
+  allItemsLabel: PropTypes.string.isRequired
+}
+
+export default MultiSelectDropdown
```

Comment:
--- $${\color{green}[AI \space Generated \space Comment]}$$ ---
The `MultiSelectDropdown` component is well-structured, but consider adding accessibility attributes for better screen reader support. For example:

```typescript
<DropdownMenuTrigger asChild>
  <Button
    variant="outline"
    className="w-full justify-between"
    aria-label={`Select ${allItemsLabel.toLowerCase()}`}
    aria-haspopup="true"
    aria-expanded={isOpen}
  >
    {/* ... */}
  </Button>
</DropdownMenuTrigger>
```

Also, consider adding keyboard navigation support for the dropdown items.